VimeoUpload/Upload/Operations/Async/MeQuotaOperation.swift

```diff
@@ -1,298 +0,0 @@
-                
-                if operation.isCancelled == true
-                {
-                    return
-                }
-                
-                // Do not check error, allow to pass [AH]
-
-                if let result = operation.result, result == false
-                {
-                    strongSelf.error = NSError.error(withDomain: UploadErrorDomain.MeQuotaOperation.rawValue, code: UploadLocalErrorCode.dailyQuotaException.rawValue, description: "Upload would exceed daily quota.")
-                }
-                else
-                {
-                    if strongSelf.avAsset != nil
-                    {
-                        strongSelf.checkApproximateWeeklyQuota() // If the asset is not nil, then we can perform the the MB-based checks
-                    }
-                    else
-                    {
-                        strongSelf.state = .finished // If the asset is nil, then it's in iCloud and we don't yet have access to the filesize
-                    }
-                }
-            })
-        }
-        
-        self.operationQueue.addOperation(operation)
-    }
-
-   private func checkApproximateWeeklyQuota()
-    {
-        let me = self.me!
-        let avAsset = self.avAsset!
-        avAsset.approximateFileSize { [weak self] (value) -> Void in
-
-            guard let strongSelf = self else
-            {
-                return
-            }
-
-            let operation = WeeklyQuotaOperation(user: me, fileSize: value)
-            operation.completionBlock = { [weak self] () -> Void in
-                
-                DispatchQueue.main.async(execute: { [weak self] () -> Void in
-                    
-                    guard let strongSelf = self else
-                    {
-                        return
-                    }
-                    
-                    if operation.isCancelled == true
-                    {
-                        return
-                    }
-                    
-                    // Do not check error, allow to pass [AH]
-
-                    if let result = operation.result, result.success == false
-                    {
-                        let userInfo = [UploadErrorKey.FileSize.rawValue: result.fileSize, UploadErrorKey.AvailableSpace.rawValue: result.availableSpace]
-                        strongSelf.error = NSError.error(withDomain: UploadErrorDomain.MeQuotaOperation.rawValue, code: UploadLocalErrorCode.weeklyQuotaException.rawValue, description: "Upload would exceed approximate weekly quota.").error(byAddingUserInfo: userInfo as [String : AnyObject])
-                    }
-                    else
-                    {
-                        strongSelf.checkApproximateDiskSpace(fileSize: value)
-                    }
-                })
-            }
-            
-            strongSelf.operationQueue.addOperation(operation)
-        }
-    }
-
-    private func checkApproximateDiskSpace(fileSize: Float64)
```

Comment:
At first I was concerned that this was removed but It looks like we do this in `ExportOperation`. I wonder why we also did this in the `MeQuotaOperation`.

Reply:
@nicolelehrer Agreed I was also confused at first. There's a lot of duplicated stuff in this framework.